custom/icds/translations/integrations/transifex.py

```diff
@@ -0,0 +1,200 @@
+        :param domain: domain name
+        :param app_id: id of the app to be used
+        :param source_lang: source lang code like en
+        :param target_lang: target lang code like hin
+        :param version: version of the app like 10475
+        :param lang_prefix: prefix if other than "default_"
+        """
+        self.app_id = app_id
+        self.domain = domain
+        self.version = int(version)
+        self.source_lang = source_lang
+        self.target_lang = target_lang
+        self.lang_prefix = lang_prefix
+        if not self.target_lang:
+            self.target_lang = self.source_lang
+        self.app_id_to_build = None
+        self.translations = OrderedDict()
+        self.headers = dict()  # headers for each sheet name
+        self.sheet_name_to_module_or_form_type_and_id = dict()
+        self.generated_files = list()
+
+    def find_build_id(self):
+        # find build id if version specified
+        if self.version:
+            from corehq.apps.app_manager.dbaccessors import get_all_built_app_ids_and_versions
+            built_app_ids = get_all_built_app_ids_and_versions(self.domain, self.app_id)
+            for app_built_version in built_app_ids:
+                if app_built_version.version == self.version:
+                    self.app_id_to_build = app_built_version.build_id
+                    break
+            if not self.app_id_to_build:
+                raise Exception("Build for version requested not found")
+        else:
+            self.app_id_to_build = self.app_id
+
+    def translation_data(self, app):
+        # get the translations data
+        from corehq.apps.app_manager.app_translations.app_translations import expected_bulk_app_sheet_rows
+        # simply the rows of data per sheet name
+        rows = expected_bulk_app_sheet_rows(app)
+
+        # get the translation data headers
+        from corehq.apps.app_manager.app_translations.app_translations import expected_bulk_app_sheet_headers
+        for header_row in expected_bulk_app_sheet_headers(app):
+            self.headers[header_row[0]] = header_row[1]
+        self.set_sheet_name_to_module_or_form_mapping(rows[u'Modules_and_forms'])
+        return rows
+
+    def set_sheet_name_to_module_or_form_mapping(self, all_module_and_form_details):
+        # iterate the first sheet to get unique ids for forms/modules
+        sheet_name_column_index = self.get_header_index(u'Modules_and_forms', 'sheet_name')
+        unique_id_column_index = self.get_header_index(u'Modules_and_forms', 'unique_id')
+        type_column_index = self.get_header_index(u'Modules_and_forms', 'Type')
+        for row in all_module_and_form_details:
+            self.sheet_name_to_module_or_form_type_and_id[row[sheet_name_column_index]] = Unique_ID(
+                row[type_column_index],
+                row[unique_id_column_index]
+            )
+
+    def get_filename(self, sheet_name):
+        return sheet_name + '_v' + str(self.version)
```

Comment:
Replied to a similar thing above https://github.com/dimagi/commcare-hq/pull/20492#discussion_r187567030
good point about conflicts. Can add a timestamp to it to avoid conflict and also a quick view of when the file was created, though that timestamp exists in the file itself as well.
Ya not needed across requests.